internal/agent/device/fileio/reader.go

```diff
@@ -30,7 +32,27 @@ func (r *Reader) PathFor(filePath string) string {
 
 // ReadFile reads the file at the provided path
 func (r *Reader) ReadFile(filePath string) ([]byte, error) {
-	return os.ReadFile(r.PathFor(filePath))
+	file, err := os.Open(r.PathFor(filePath))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+	var data []byte
+	for {
+		buf := make([]byte, 1024) // TODO: benchmark and optimize buffer size
```

Comment:
I'm probably missing some information about why this is better.
1. Allocating a small buffer, reading a small chunk, and copying it into a larger buffer can't be faster than reading the whole thing in one shot (more system calls, data copies, etc).
2. In either case, we are allocating a buffer containing the full contents of the file (`data` here), so how does it reduce RSS?

Reply:
haha yeah I will add benchmarks before re add. creating the buffer in the for loop is a bug anyways.